server/lib/pbench/test/unit/test_process_tb.py

```diff
@@ -1,67 +1,75 @@
+import os
 import logging
 import pytest
 
-from pbench import get_pbench_logger
+from pathlib import Path
+
 from pbench.process_tb import ProcessTb, Results
-from pbench.test import tarball, bad_tarball, receive_dir
+from pbench.test import tarball, bad_tarball, receive_dir, fixture_path
 
 
 class MockConfig:
-    def __init__(self):
-        self.TS = "run-1970-01-02T00:00:00.000000"
+    TS = "run-1970-01-02T00:00:00.000000"
 
     def get(self, section: str, option: str):
         pass
 
 
 class TestProcessTb:
     @pytest.fixture(autouse=True)
-    def config_and_logger(self, valid_config):
+    def logger(self, test_logger):
         # Setup the configuration and logger
-        global tmp
-        config, tmp = valid_config
-        self.logger = get_pbench_logger("pbench", config)
+        self.logger = test_logger
 
     def test_get_receive_dir_failed(self, monkeypatch):
         def mock_get(section: str, option: str):
             if option == "pbench-receive-dir-prefix":
                 return ""
+            assert False
 
         expected_error_msg = "Failed: No value for config option pbench-receive-dir-prefix in section pbench-server"
         monkeypatch.setattr(MockConfig, "get", mock_get)
         with pytest.raises(ValueError) as e:
             ProcessTb(MockConfig, self.logger)
 
-        assert str(e).endswith(expected_error_msg)
+        assert expected_error_msg in str(e)
 
     def test_get_receive_dir_value_failed(self, monkeypatch):
         def mock_get(section: str, option: str):
             if option == "pbench-receive-dir-prefix":
                 return receive_dir / "wrong-directory"
+            assert False
 
         expected_error_path = str(receive_dir / "wrong-directory-002")
         monkeypatch.setattr(MockConfig, "get", mock_get)
         with pytest.raises(Exception) as e:
             ProcessTb(MockConfig, self.logger)
 
-        assert str(e).endswith(expected_error_path)
+        assert expected_error_path in str(e)
 
     def test_process_tb_file_not_found_error(self, monkeypatch, caplog):
         """checks normal processing when tar ball is not present"""
-        tb_path = tmp / receive_dir / "bad_tarball"
+        tb_path = fixture_path / "bad_tarball"
+
+        def mock_glob(*args, **kwargs):
+            tb_path = fixture_path / "bad_tarball-002" / "bad_log.tar.xz.md5"
+            return [tb_path]
 
         def mock_get(self, section: str, option: str):
             if option == "pbench-receive-dir-prefix":
                 return tb_path
             if option == "put-token":
                 return "Authorization-token"
+            assert False
 
         def mock_results_push(self, ctrl, tb, token):
             if str(tb).startswith(str(tb_path)):
                 raise FileNotFoundError(f"No such file or directory: '{tb.name}'")
+            assert False
```

Comment:
`mock_glob()` is overwriting the value in `tb_path`...is that really what you want to do?

I think we want to set up the value (i.e., in an outer-scope local variable) returned by `mock_get()` for the prefix, and then we want to confirm that that value is passed into `mock_glob()` (via its `self` parameter); then we want `mock_results_push()` to be able to verify that the `tb` value that it receives came from the list that `mock_glob()` returned, so that list should probably be a second outer-scope local variable, and it should probably be constructed based on the value in the first local variable.

Does that all make sense?

Reply:
I think that this should be the general pattern for each of the test cases.  For the cases where control is supposed to exit prematurely, the mocks later in the sequence should fail the test if they are called.